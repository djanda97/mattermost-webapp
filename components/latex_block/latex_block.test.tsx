// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import LatexBlock from 'components/latex_block/latex_block';

describe('components/LatexBlock', () => {
    const defaultProps = {
        content: '```latex e^{i\\pi} + 1 = 0```',
        enableLatex: true,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<LatexBlock {...defaultProps}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
