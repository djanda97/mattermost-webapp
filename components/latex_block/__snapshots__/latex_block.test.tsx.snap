// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LatexBlock error in katex 1`] = `
<div
  className="post-body--code tex"
>
  \`\`\`latex e^{i\\pi + 1 = 0\`\`\`
</div>
`;

exports[`components/LatexBlock latex is disabled 1`] = `
<div
  className="post-body--code tex"
>
  \`\`\`latex e^{i\\pi} + 1 = 0\`\`\`
</div>
`;

exports[`components/LatexBlock should match snapshot 1`] = `
<div
  className="post-body--code tex"
>
  \`\`\`latex e^{i\\pi} + 1 = 0\`\`\`
</div>
`;
